test/runtests.jl

```diff
@@ -72,4 +73,73 @@ include("../docs/src/ferrite-examples/heat-equation.jl")
+                            #(7,Hexahedron, Lagrange{3,RefHexahedron,2}()^3)
+                        ]
+            dim = Ferrite.getdim(ip)
+            @testset "$component" for component ∈ 1:dim
+                # Compute solution
+                dh, u = manufactured_linear_elastic_problem(geo, ip, size, component)
+                qr = QuadratureRule{Ferrite.getrefshape(ip)}(1)
+                ip_geo = Ferrite.default_interpolation(typeof(Ferrite.getcells(FerriteViz.get_grid(dh), 1)))
+
+                # Check solution
+                cellvalues = CellValues(qr, ip, ip_geo);
+                for cell in CellIterator(dh)
+                    reinit!(cellvalues, cell)
+                    n_basefuncs = getnbasefunctions(cellvalues)
+                    coords = getcoordinates(cell)
+                    uₑ = u[celldofs(cell)]
+                    for q_point in 1:getnquadpoints(cellvalues)
+                        x = spatial_coordinate(cellvalues, q_point, coords)
+                        for i in 1:n_basefuncs
+                            uₐₙₐ    = prod(cos, x*π/2)
+                            uₐₚₚᵣₒₓ = function_value(cellvalues, q_point, uₑ)[component]
+                            @test isapprox(uₐₙₐ, uₐₚₚᵣₒₓ; atol=1e-2)
+                        end
+                    end
+                end
+
+                # Compute gradient/flux field
+                (dh_grad, u_grad) = FerriteViz.interpolate_gradient_field(dh, u, :u)
+
+                # Check gradient of solution
+                # cellvalues_grad = CellValues(qr, Ferrite.getfieldinterpolation(dh_grad, :gradient), ip_geo);
+                # for cell in CellIterator(dh_grad)
+                #     reinit!(cellvalues_grad, cell)
+                #     n_basefuncs = getnbasefunctions(cellvalues_grad)
+                #     coords = getcoordinates(cell)
+                #     uₑ = u_grad[celldofs(cell)]
+                #     for q_point in 1:getnquadpoints(cellvalues_grad)
+                #         x = spatial_coordinate(cellvalues_grad, q_point, coords)
+                #         for i ∈ 1:n_basefuncs
+                #             uₐₚₚᵣₒₓ = function_value(cellvalues_grad, q_point, uₑ)
+                #             for d ∈ 1:dim
+                #                 uₐₙₐ = π/2
+                #                 for j ∈ 1:(d-1)
+                #                     uₐₙₐ *= cos(x[j]*π/2)
+                #                 end
+                #                 uₐₙₐ *= -sin(x[d]*π/2)
+                #                 for j ∈ (d+1):dim
+                #                     uₐₙₐ *= cos(x[j]*π/2)
+                #                 end
+                #                 @test isapprox(uₐₙₐ, uₐₚₚᵣₒₓ[d]; atol=1e-1)
+                #             end
+                #         end
+                #     end
+                # end
+            end
+        end
+    end
+end
```

Comment:
I guess we should split this and move the script into https://github.com/Ferrite-FEM/Ferrite.jl/pull/640 and use `apply_analytical` and a point-wise evaluation of the gradients here.